agent/agent_test.go

```diff
@@ -4705,3 +4712,64 @@ func TestSharedRPCRouter(t *testing.T) {
 	require.NotNil(t, mgr)
 	require.NotNil(t, server)
 }
+
+func TestAgent_ListenHTTP_MultipleAddresses(t *testing.T) {
+	caConfig := tlsutil.Config{}
+	tlsConf, err := tlsutil.NewConfigurator(caConfig, hclog.New(nil))
+	require.NoError(t, err)
+	bd := BaseDeps{
+		Deps: consul.Deps{
+			Logger:          hclog.NewInterceptLogger(nil),
+			Tokens:          new(token.Store),
+			TLSConfigurator: tlsConf,
+		},
+		RuntimeConfig: &config.RuntimeConfig{
+			HTTPAddrs: []net.Addr{
+				&net.TCPAddr{IP: net.ParseIP("127.0.0.1")},
+				&net.TCPAddr{IP: net.ParseIP("127.0.0.1")},
```

Comment:
why aren't these different ip addresses?

Reply:
I guess I could use another 127.0.0.0/24 address? The IP is the same, but the port is different (because it is set to 0 and the OS will choose one), so we end up with two different listener addresses. I could also add `Port: 0` to make that more explicit.

Reply:
Actually if that's the case you should probably use freeport to grab two ports officially at the top of the test and use them here. Unit tests cannot actually ever rely upon `:0` port grabbing because that invalidates some tests that rely upon a particular port being reserved _and not listening_ for a portion of those tests.